tests/test_cache_api.py

```diff
@@ -193,6 +195,56 @@ def c(b):
     flow.get("c")
 
 
+def test_delete_artifact_with_multiple_metadata_files(builder):
+    builder.assign("a", 1)
+
+    @builder
+    @bn.memoize(False)
+    def b(a):
+        return 2
+
+    # Define `c` several times, each with non-functional version differences. This means
+    # each defintion will have a new metadata entry but the same artifact, so we'll have
+    # many entries pointing to the same artifact.
+    c_entries = []
+    for i in range(4):
+
+        @builder
+        @bn.memoize(False)
+        @bn.version(minor=i)
+        def c(b):
+            return b + 1
+
+        flow = builder.build()
+        flow.get("c")
+
+        (c_entry,) = [
+            entry
+            for entry in flow.cache.get_entries()
+            if entry.entity == "c" and entry not in c_entries
+        ]
+        for old_c_entry in c_entries:
+            assert c_entry.metadata_url != old_c_entry.metadata_url
+            assert c_entry.artifact_url == old_c_entry.artifact_url
+        c_entries.append(c_entry)
+
+    # This deletes the artifact and returns True.
+    assert c_entries[0].delete()
+    # The artifact is already deleted, so this returns False.
+    assert not c_entries[1].delete()
+
+    # This should attempt to load the last entry, detect that it's missing, and
```

Comment:
nit: You can be more explicit here that it refers to the artifact

Reply:
Good point!